src/checkers/Cashflow.ml

```diff
@@ -1604,21 +1630,38 @@ module ScillaCashflowChecker
                   cf_tag_expr e (lub_tags (get_id_tag f) NoInfo) (AssocDictionary.make_dict ()) (AssocDictionary.make_dict ()) ctr_tag_map in
              AssocDictionary.insert (get_id f) e_tag acc_env)
       in
-      let rec tagger cconstraint components param_env field_env ctr_tag_map =
+    let rec tagger cconstraint (fields:(field list)) components param_env field_env ctr_tag_map =
         let (new_constraint, tmp_param_env, _, tmp_ctr_tag_map, constraint_changes) =
           cf_tag_expr cconstraint NotMoney param_env (AssocDictionary.make_dict()) ctr_tag_map in
-        let (new_ts, new_param_env, new_field_env, tmp_ctr_tag_map, changes) =
-          List.fold_right components ~init:([], tmp_param_env, field_env, tmp_ctr_tag_map, constraint_changes) 
+
+      (* adding field tagging *)
+      let (new_fs, new_param_env1, new_field_env1, tmp_ctr_tag_map, field_changes) =
```

Comment:
I don't mind the renaming of `new_xxx_env` as such, but the current naming convention is somewhat inconsistent. `tmp_ctr_tag_map` should be named `new_ctr_tag_map1` here.

Reply:
I also wouldn't mind if `field_changes` was renamed to reflect that it represents changes in either the constraints or the field initialisers.